tools/chapel-py/src/core-types-pernode.cpp

```diff
@@ -75,19 +76,53 @@ static const char* intentToString(IntentType intent) {
   return qualifierToString(Qualifier(int(intent)));
 }
 
-static const resolution::ResolvedExpression*
-scopeResolveResultsForNode(Context* context, const AstNode* node) {
+static const ID scopeResolveVisibilityStmt(Context* context, const AstNode* visibilityStmt, const AstNode* node) {
+  if (visibilityStmt->isUse() || visibilityStmt->isImport()) {
+    auto useParent = parsing::parentAst(context, visibilityStmt);
+    auto scope = resolution::scopeForId(context, useParent->id());
+    auto reScope = resolution::resolveVisibilityStmts(context, scope);
+    for (auto visCla: reScope->visibilityClauses()) {
+      if(visCla.visibilityClauseId().contains(node->id())) {
+        return visCla.scope()->id();
+      }
+    }
+  }
+  return ID();
+}
+
+static const ID scopeResolveFunction(Context* context, const AstNode* fnNode, const AstNode* node) {
+  if (auto fn = fnNode->toFunction()) {
+    auto byId = resolution::scopeResolveFunction(context, fn->id())->resolutionById();
```

Comment:
Consider putting `->resolutionById()`; on a new line to avoid excessive line size.